### PR TITLE
Fix windows native build with RubyInstaller and DevKit

### PR DESCRIPTION
**Pull request by larskanis**
This depends on https://github.com/flavorjones/mini_portile/pull/51 in regards to the handling of configure_options. I therefore assigned a git link in the Gemfile, temporary.

Rubys stdlib shellwords is generelly an issue on Windows. It's sadly, but Windows uses several different ways to escape command line options. While shellwords mostly works in Msys bash (and therefore in a Makefile), it is completely wrong in cmd.exe or msvcrt based programs. It's annoying, but escape rules in mkmf.rb functions (like try_compile etc) are different to that of $CPPFLAGS etc. which run in a Makefile. I made an attempt to codify the different escape rules in, but don't think this is necessary for nokogiri:
   https://github.com/larskanis/shellwords/tree/master/lib/shellwords

As a compromise I used String#inspect for the lists of patches. This works well enough on all platforms.

This fixes https://github.com/sparklemotion/nokogiri/issues/1190 and already includes  https://github.com/sparklemotion/nokogiri/pull/1308 , to make sure, that the native build is actively checked.

### CODE REVIEW

**Comment by flavorjones** [comment]
Thanks for submitting this. I need to dig in an understand what's going on here, and why the builds are failing.

**Comment by larskanis** [comment]
I'll check why ruby-1.9.3 on OSX failed and update the PR if necessary.

**Comment by larskanis** [comment]
@flavorjones Bummer! Travis-CI currently doesn't accept OSX requests and I don't have a Mac at hand. So I currently can only find out why ruby-1.9.3 on OSX fails, by temporary adding another commit to this PR, to debug this.

**Comment by larskanis** [comment]
Some configure_options were still quoted. This is fixed now.

File: ext/nokogiri/extconf.rb [hunk 1]
```diff
@@ -526,8 +529,8 @@ def install
       }
 
       # Defining a macro that expands to a C string; double quotes are significant.
-      $CPPFLAGS << ' ' << "-DNOKOGIRI_#{recipe.name.upcase}_PATH=\"#{recipe.path}\"".shellescape
-      $CPPFLAGS << ' ' << "-DNOKOGIRI_#{recipe.name.upcase}_PATCHES=\"#{recipe.patch_files.map { |path| File.basename(path) }.join(' ')}\"".shellescape
+      $CPPFLAGS << ' ' << "-DNOKOGIRI_#{recipe.name.upcase}_PATH=\"#{recipe.path}\"".inspect
+      $CPPFLAGS << ' ' << "-DNOKOGIRI_#{recipe.name.upcase}_PATCHES=\"#{recipe.patch_files.map { |path| File.basename(path) }.join(' ')}\"".inspect
```
Comment (ccoenen) [inline ext/nokogiri/extconf.rb]: is the `.inspect` intentional? seems a little odd to me?

Comment (larskanis) [inline ext/nokogiri/extconf.rb]: It is. This is what I refer in the PR description "As a compromise I used String#inspect for the lists of patches. This works well enough on all platforms."

**Comment by marutosi** [comment]
What is status of this PR?
What is blocker in https://github.com/flavorjones/mini_portile/pull ?

**Comment by flavorjones** [comment]
Hi all, I just emailed nokogiri-talk about a Windows release. If you don't subscribe, you should, but here's the link in any case:

> https://groups.google.com/d/msg/nokogiri-talk/aJkGZz5bhRo/3yu3BNFPDgAJ

TL;DR we're going to get this out the door this week, honest. Thanks for your patience.

**Comment by Azolo** [comment]
Nice! Let me know if you need anything from RubyInstaller.

**Comment by larskanis** [comment]
@Azolo Good to know! For now, there's everything mighty fine with RubyInstaller in respect to nokogiri.

**Comment by flavorjones** [comment]
Merging. Will be on master shortly.
